packages/explat-client/src/create-explat-client.ts

```diff
@@ -0,0 +1,158 @@
+/**
+ * Internal dependencies
+ */
+import { ExperimentAssignment, Config } from './types';
+import * as ExperimentAssignments from './internal/experiment-assignments';
+import * as Request from './internal/requests';
+import * as State from './internal/state';
+import * as Timing from './internal/timing';
+import * as Validation from './internal/validations';
+
+/**
+ * The number of seconds before we abandon fetching an experiment
+ */
+const EXPERIMENT_FETCH_TIMEOUT = 5000;
+
+export interface ExPlatClient {
+	/**
+	 * Loads and returns an Experiment Assignment Promise, starting an assignment if necessary.
+	 *
+	 * Call as many times as you like, it will only make one request at a time (per experiment) and
+	 * will only trigger a request when the assignment TTL is expired.
+	 *
+	 * Will never throw in production, it will return the default assignment.
+	 * It should not be run on the server but it won't crash anything.
+	 *
+	 * @param experimentName The experiment's name
+	 */
+	loadExperimentAssignment: ( experimentName: string ) => Promise< ExperimentAssignment >;
+
+	/**
+	 * Get an already loaded Experiment Assignment, will throw if there is an error, e.g. if it hasn't been loaded.
+	 *
+	 * Make sure loadExperimentAssignment has been called before calling this function.
+	 *
+	 */
+	dangerouslyGetExperimentAssignment: ( experimentName: string ) => ExperimentAssignment;
+}
+
+export class MissingExperimentAssignmentError extends Error {
+	constructor( ...params: any[] ) {
+		super( ...params );
+
+		// Maintains proper stack trace for where our error was thrown (only available on V8)
+		if ( Error.captureStackTrace ) {
+			Error.captureStackTrace( this, MissingExperimentAssignmentError );
+		}
+
+		this.name = 'MissingExperimentAssignmentError';
+	}
+}
+
+/**
+ * Create an ExPlat Client
+ *
+ * @param config Configuration object
+ */
+export default function createExPlatClient( config: Config ): ExPlatClient {
+	/**
+	 * The null experiment assignment we return when we can't retrieve one.
+	 */
+	const createNullExperimentAssignment = (): ExperimentAssignment => ( {
+		experimentName: 'null_experiment_assignment',
```

Comment:
> Things might get weird since this is a legal name. Is there a reason not to populate it with the actual name?
>
> Also, what's the rationale behind creating an "empty" object rather than using null

Yeah I was thinking of this, I might add the name in. The rationale here is that `loadExperimentAssignment` for simplicity of use always returns an `ExperimentAssignment` (less cases for an experimenter to deal with), and I wanted it to be fairly explicit that this happens, but maybe I will add the name in and add an extra property indicating that it is a "null assignment".